internal/js_printer/js_printer.go

```diff
@@ -3386,6 +3429,52 @@ func (p *printer) shouldIgnoreSourceMap() bool {
 	return true
 }
 
+func (p *printer) printHotModuleReloadPrelude(s *js_ast.SFunction) {
+	filePath := s.HotModule.FilePath
+	name := p.renamer.NameForSymbol(s.HotModule.RefreshFn)
+	contextVarName := "___spinx_url_" + name + "__"
+	hotVarName := "___spinx_hot_" + name + "__"
+	refreshSigVarName := "__spinx_refresh_sig_" + name + "__"
+
+	p.print(fmt.Sprintf("console.log('This file has HMR enabled for lol \\'%s\\'');\n", name))
+
+	p.print(`
+var ` + contextVarName + ` = "` + filePath + `";
```

Comment:
🙈

Reply:
A disadvantage of printing this boilerplate is that it skips minification. But:
* You'd always be disabling hmr when minification counts (production)
* It's adding very compressible, repeating text